server/controller/rides.js

```diff
@@ -6,20 +6,16 @@ class RidesController{
             if (err) return next(err);
             res.status(200).json({
                 success: true,
-                message: 'All Rides has been retrieve.',
-                data: data.rows
+                message: 'All rides has been retrieved.',
+                rides: data.rows
             })
         })  
     }
     static getARide(req, res, next){
```

Comment:
I don't understand why you had to do getARide like this. I think when you select all where the id is what you provided in the param, the data that is going to be in the row would be the one that has the id you queried with, why do you have to loop again to check if the id is the same as you passed? the check that is supposed to have happened here is if data.row is not empty return the result else, return no ride found and then you return err is there is err

Reply:
I have fix this as well.